### PR TITLE
feat(paper-review): wire onPreviewDiff to render the proposal diff inline

### PR DESCRIPTION
**Pull request by Chris0Jeky**
## Summary

De-stubs the `onPreviewDiff` handler in the Paper deep-review surface (`PaperReviewView.vue`) — a Wave-2 activation slice. Previously the handler only fired a placeholder toast (`Preview diff is not wired yet; no diff was loaded.`). It now loads the real proposal diff and renders it **inline** in the review surface.

- Calls `automationApi.getProposalDiff(id)` (GET `/automation/proposals/{id}/diff`) and renders the returned diff text inline, below the change section — **no drawer/modal** (inline is the settled UX decision, consistent with how provenance and side-effects render in the same surface).
- Behaviour mirrors the legacy `useReviewActions.handleToggleDiff` pattern:
  - **Toggle off** when invoked again for the same proposal (no re-fetch).
  - **Stale-response guard** via a monotonic `latestDiffRequestId` counter — a slow response for a since-changed proposal is dropped.
  - **Clears preview state** when the active proposal changes (a `watch` on the active proposal id).
  - **Error handling**: `toast.error(getErrorDisplay(e, 'Failed to load proposal diff').message)` and tears the inline surface back down — matching how other handlers in this file/area surface errors.
  - **Empty-diff state**: an explicit "No changes to preview" message when the diff string is empty (the legacy `&& selectedDiff` truthy gate would have shown nothing).
- The diff `<pre>` block is styled with Paper deep-review tokens (`--mono`, `--line-soft`, `--ink-2`, `tk-serial`/`tk-h3`/`tk-meta`) so it sits consistently in the surface. Trigger is the existing **Space** keymap shortcut.
- Out of scope (untouched): `onDefer` and `onReportBadSuggestion` — those need net-new backend and are separate slices.

## Test Plan

Extended `src/tests/views/paper/review/PaperReviewView.spec.ts`. The obsolete stub-toast assertion was replaced with four cases:

- **loads and renders the proposal diff inline when preview diff is invoked** — Space calls `getProposalDiff` with the active proposal id and renders the returned diff in the inline `<pre>`; no stub toast.
- **hides the inline diff when preview diff is invoked again for the same proposal** — second Space toggles the diff off and does not re-fetch.
- **renders an empty-diff state when the proposal has no changes to preview** — an empty diff string renders the "No changes to preview" state instead of an empty `<pre>`.
- **surfaces a toast and does not crash when the diff request fails** — a rejected request fires `toast.error` and leaves the view mounted with no stale diff region.

Verification (from `frontend/taskdeck-web`):
- `npm run typecheck` — pass (no errors).
- `npx vitest --run src/tests/views/paper/review/PaperReviewView.spec.ts` — **35 passed** (4 new).
- `npx eslint` on the two changed files — clean.

### CODE REVIEW

**Comment by Chris0Jeky** [comment]
## Adversarial self-review

Reviewed the full diff for the `onPreviewDiff` de-stub. Verified against the legacy `useReviewActions.handleToggleDiff` reference and walked the async/edge-case paths. Findings by severity:

### CRITICAL / HIGH
None found.

### MEDIUM
- **Stale-response correctness (verified, no fix needed).** Two clearing mechanisms coexist: the `watch(activeProposal.id)` clears + bumps `latestDiffRequestId` on proposal change, and the in-handler guard re-checks `requestId !== latestDiffRequestId || previewDiffProposalId !== p.id` after the `await`. Traced the A-in-flight → switch-to-B race: the watcher bumps the counter so A's late response is dropped, and B never inherits A's diff. Correct, no double-render or leak.
- **Loading flag never sticks (verified).** Toggling off mid-flight (Space pressed twice fast) sets `previewDiffLoading=false` and bumps the request id in the toggle-off branch; the in-flight `finally` only clears loading when `requestId === latestDiffRequestId`, so it won't resurrect a stale spinner.

### LOW
- **Empty-diff UX divergence from legacy (intentional).** Legacy gates rendering on `&& selectedDiff` (truthy), so an empty diff string renders nothing. This PR intentionally renders an explicit "No changes to preview" state for `''`, per the slice spec. Covered by a test.
- **`getProposalDiff` not de-duplicated across rapid presses for *different* proposals.** Switching proposals quickly fires a new request each time; older ones are dropped by the stale guard rather than aborted. Matches legacy behaviour (no AbortController there either); acceptable for this surface. Not changing to keep scope tight and parity with legacy.

### Out of scope (untouched, no issue seeded — already tracked)
- `onDefer` and `onReportBadSuggestion` remain stubs (need net-new backend; explicitly out of scope per the slice and already TODO-tracked under #1002).

### Verification
- `npm run typecheck` — pass.
- `npx vitest --run src/tests/views/paper/review/PaperReviewView.spec.ts` — 35 passed (4 new: inline render, same-proposal toggle-off, empty-diff state, API-error toast without crash).
- `npx eslint` on both changed files — clean.
- CI: Paper Color Audit, Docs Governance, Secret Scan green (color audit confirms the diff `<pre>` styling uses approved Paper tokens only).

No code changes required from this review — all traced edge cases behave correctly and are test-covered.

_Note: the gemini-code-assist bot comment above is a daily-quota notice, not a finding._

File: frontend/taskdeck-web/src/views/paper/PaperReviewView.vue [hunk 1]
```diff
@@ -597,10 +625,39 @@ function onToggleProvenance() {
...
+    return
+  }
+
+  const requestId = ++latestDiffRequestId
+  previewDiffProposalId.value = p.id
+  previewDiff.value = null
+  previewDiffLoading.value = true
+
+  try {
+    const diff = await automationApi.getProposalDiff(p.id)
```
[the remark this answers is not in the document]
Comment (Chris0Jeky) [reply frontend/taskdeck-web/src/views/paper/PaperReviewView.vue:674]: Fixed in 33ef6d60 — `onPreviewDiff` now guards no-op proposals before fetching: if `!proposal.diffPreview && proposal.operations.length === 0` (the exact condition that makes `GetProposalDiffAsync` return 404 at `AutomationProposalService.cs:358`), it shows the empty-diff state directly without firing the request. Belt-and-suspenders: the catch also treats a `NotFound` (404) `errorCode` as the empty state rather than an error toast. New test: 'shows the empty-diff state without fetching for a no-operation proposal'.

[the remark this answers is not in the document]
Comment (Chris0Jeky) [reply frontend/taskdeck-web/src/views/paper/PaperReviewView.vue:674]: Fixed in 33ef6d60 (mitigation) + tracked as **#1235** (real backend fix). You're right: `/diff` returns the original proposal's operations while Apply materializes the latest `ProposalRevision` (`AutomationExecutorService.MaterializeEffectiveProposalAsync`), so a previewed diff can mislead when a saved revision exists. Since revision-aware diff is a backend change (out of this frontend slice's scope), I seeded **#1235** for it and added a UI **caveat** here: when `revisionCount > 0`, the preview surface shows '⚠ This preview shows the original proposal. A saved edit (revision) will be applied instead…'. New test: 'shows a revision caveat in the diff preview when a saved revision exists'.

**Comment by Chris0Jeky** [comment]
## Fix evidence (Codex review)
Both P2 correctness findings fixed in `33ef6d60` (37/37 spec tests pass, typecheck + eslint clean):
- **No-op proposals → 404** — `onPreviewDiff` now short-circuits when `!diffPreview && operations.length === 0` (the exact backend 404 condition) to show the empty-diff state without fetching; the catch also maps a `NotFound` errorCode to the empty state. +1 test.
- **Preview vs effective revision** — added a UI caveat when `revisionCount > 0` (preview shows the original; the saved revision applies). The real backend fix (revision-aware `/diff`) is tracked in **#1235**. +1 test.

**Comment by Chris0Jeky** [comment]
## Fix evidence (Codex round 2)
Both fixed in `ecd9ff47` (38/38 spec tests, typecheck + eslint clean):
- **Visibility** — the diff now scrolls into view when opened (`scrollDiffIntoView()` on both the fetched and no-op paths; jsdom-guarded), so Space shows the diff even on tall review surfaces.
- **Over-broad 404** — removed the catch's `NotFound → empty` mapping. The no-op case is handled by the pre-fetch guard, so a fetch-time 404 now means the proposal was deleted/dismissed elsewhere and correctly surfaces an error (+ a test asserting a 404 on an operations-bearing proposal errors, not empties).
